### DISCUSSION TITLE
CustomField @entangle()

### DISCUSSION THREAD

**Discussion by RPotkowaVegasystems** [filamentphp/filament#9160] [Help]
I have a custom Field, with additional public properties in my "class CustomField extends Field".

```
public string myProperty = 'foo';

```

How can i @entangle it in my x-data?

```
<div x-data="{
        state: $wire.{{ $applyStateBindingModifiers("\$entangle('{$getStatePath()}')") }},
        var: $wire.$entangle('myProperty'),
     }">
````
Is not working.

**Comment by leandrocfe** [comment]
This?

CustomField class
```php
protected string | null $myProperty = null;

public function myProperty(string | null $text): static
{
    $this->myProperty = $text;

    return $this;
}

public function getMyProperty(): string | null
{
    return $this->evaluate($this->myProperty);
}
```
custom-field blade
```html
<div x-data="{ state: $wire.$entangle('{{ $getStatePath() }}'), property: '{{ $getMyProperty() }}' }">
    <span x-text="property"></span>
</div>
```
Field
```php
CustomField::make('field')
    ->myProperty('new value')
```

**Reply by RPotkowaVegasystems** [reply]
Ok, but is it works also, when the value $getMyProperty() is dynamic and changes on live()??? 

I did it before like you to create a list x-for, and when the values changes in my property, i got "Uncaught TypeError: Illegal invocation" in my x-for list.

My Problem is, i would like to create a TextInput Searchfield with a dropdown suggestions under the Field itself, when i type something in and get some search results.

**Reply by leandrocfe** [reply]
[live()](https://filamentphp.com/docs/3.x/forms/fields/custom#obeying-state-binding-modifiers)
